### ISSUE TITLE
IR Failing After GPUMultiBuffering Failed (iree-gpu-multi-buffering).

### ISSUE THREAD

**Issue by pashu123** [iree-org/iree#11520]
print-ir-before-all: https://gist.github.com/pashu123/c9c01d1397a820dad48ecdd01e9ae5d5

```
failed to run translation of source executable to target executable for backend #hal.executable.target<"vulkan", "vulkan-spirv-fb", {spirv.target_env = #spirv.target_env<#spirv.vce<v1.6, [Shader, Float64, Float16, Int64, Int16, Int8, StorageBuffer16BitAccess, StorageUniform16, StoragePushConstant16, StorageBuffer8BitAccess, UniformAndStorageBuffer8BitAccess, StoragePushConstant8, GroupNonUniform, GroupNonUniformVote, GroupNonUniformArithmetic, GroupNonUniformBallot, GroupNonUniformShuffle, GroupNonUniformShuffleRelative, GroupNonUniformClustered, GroupNonUniformQuad, VariablePointers, VariablePointersStorageBuffer, CooperativeMatrixNV], [SPV_KHR_16bit_storage, SPV_KHR_8bit_storage, SPV_KHR_storage_buffer_storage_class, SPV_KHR_variable_pointers, SPV_NV_cooperative_matrix]>, api=Vulkan, AMD:DiscreteGPU, #spirv.resource_limits<max_compute_shared_memory_size = 65536, max_compute_workgroup_invocations = 1024, max_compute_workgroup_size = [1024, 1024, 1024], subgroup_size = 64, min_subgroup_size = 32, max_subgroup_size = 64, cooperative_matrix_properties_nv = [#spirv.coop_matrix_props<m_size = 16, n_size = 16, k_size = 16, a_type = f16, b_type = f16, c_type = f16, result_type = f16, scope = <Subgroup>>]>>}>
    %4 = linalg.generic {indexing_maps = [#map, #map, #map], iterator_types = ["parallel", "parallel", "parallel"]} ins(%3, %0 : tensor<10x4096x4096xf16>, tensor<10x4096x4096xf16>) outs(%0 : tensor<10x4096x4096xf16>) {
         ^
```

### Steps to reproduce your issue

Repro IR: https://gist.github.com/pashu123/4e8d358a9ad28cb547192fe32cc9f8e5 
`iree-compile  --iree-input-type=none --iree-vm-bytecode-module-output-format=flatbuffer-binary --iree-hal-target-backends=vulkan  --mlir-print-debuginfo --mlir-print-op-on-diagnostic=false --iree-llvm-target-cpu-features=host --iree-stream-resource-index-bits=64 --iree-vm-target-index-bits=64 --iree-util-zero-fill-elided-attrs -iree-vulkan-target-triple=rdna3-unknown-unknown --iree-flow-enable-padding-linalg-ops --iree-flow-linalg-ops-padding-size=32 --iree-flow-enable-conv-img2col-transform test.mlir`

### What component(s) does this issue relate to?

Compiler

**Comment by ThomasRaoux** [comment]
We need to remove the trivial loops before doing the multi-buffering.

**Comment by qedawkins** [comment]
The failure in multi-buffering is fixed by #11523, but it looks like this example fails later as the fused elementwise op doesn't get distributed.

**Comment by qedawkins** [comment]
it looks like there is a bug with the input IR, namely the fused elementwise op is taking an empty tensor as input
```mlir
    %0 = tensor.empty() : tensor<10x4096x4096xf16> loc(#loc6)
    %4 = linalg.generic {indexing_maps = [#map, #map, #map], iterator_types = ["parallel", "parallel", "parallel"]} ins(%3, %0 : tensor<10x4096x4096xf16>, tensor<10x4096x4096xf16>) outs(%0 : tensor<10x4096x4096xf16>) {
    ^bb0(%in: f16 loc("/home/prashant/test.mlir":15:10), %in_1: f16 loc("/home/prashant/test.mlir":15:20), %out: f16 loc("/home/prashant/test.mlir":15:32)):
      %5 = arith.mulf %in_1, %cst : f16 loc(#loc18)
      %6 = arith.addf %in, %5 : f16 loc(#loc19)
      linalg.yield %6 : f16 loc(#loc20)
    } -> tensor<10x4096x4096xf16> loc(#loc14)
```

**Comment by benvanik** [comment]
(good catch - `tensor.empty` values are good in the `outs` but invalid in `ins`)

**Comment by qedawkins** [comment]
Closing as this is fixed by #11523.
